### PR TITLE
add dashboard flag utils tests and adjust some code

### PR DESCRIPTION
**Pull request by marviniter**
# Description

Adding integration test for dashboard '--address' flag and adjust some code

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by marviniter** [comment]
@artursouza CC

File: utils/utils.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func WriteTable(writer io.Writer, csvContent string) {
 
 func TruncateString(str string, maxLength int) string {
 	strLength := len(str)
-	if strLength <= maxLength {
+	if strLength <= maxLength || maxLength <= 3 {
```
Comment (artursouza) [inline utils/utils.go:64]: What is this change about?

File: utils/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package utils
```
Comment (artursouza) [inline utils/utils_test.go:1]: This is unit test, E2E tests sit here: https://github.com/dapr/cli/tree/master/tests/e2e

The tests I was asking for are the E2E tests, where the CLI commands is actually executed for real.

**Review by artursouza — CHANGES_REQUESTED** [review]

File: utils/utils.go (hunk 1 above)
Comment (marviniter) [reply utils/utils.go:64]: I write the unit test,when maxLength is 0,will cause panic: 
runtime error: slice bounds out of range [:-3] [recovered]
	panic: runtime error: slice bounds out of range [:-3]
So add this condition.

File: utils/utils_test.go (hunk 2 above)
Comment (marviniter) [reply utils/utils_test.go:1]: Yes this PR just for unit test, i'll submit new PR for E2E tests later.

Comment (marviniter) [reply utils/utils_test.go:1]: Or should i add E2E tests in this PR?

File: pkg/version/version_test.go [hunk 3]
```diff
@@ -114,7 +114,7 @@ func TestGetVersionsGithub(t *testing.T) {
 	t.Run("error on bad addr", func(t *testing.T) {
 		version, err := GetLatestReleaseGithub("http://a.super.non.existant.domain/")
 		assert.Equal(t, "", version)
-		assert.EqualError(t, err, "Get \"http://a.super.non.existant.domain/\": dial tcp: lookup a.super.non.existant.domain: no such host")
+		assert.Containsf(t, err.Error(), "Get \"http://a.super.non.existant.domain/\": dial tcp: lookup a.super.non.existant.domain", ": no such host")
```
Comment (marviniter) [inline pkg/version/version_test.go:117]: I run the test and got the error message "dial tcp: lookup a.super.non.existant.domain on 192.168.0.1:53: no such host" so change the assert condition.

**Comment by mukundansundar** [comment]
@marviniter Can you resolve the conflicts?

**Comment by mukundansundar** [comment]
@artursouza Can you review this PR?

**Comment by marviniter** [comment]
> @marviniter can you resolve the PR conflicts?

ok,i will solve it later.

**Comment by mukundansundar** [comment]
@marviniter Can you resolve the PR conflicts ?

**Comment by mukundansundar** [comment]
for now closing this PR. Please open a new PR for this fix.

**Pull request closed without merging**